src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -1041,4 +1049,41 @@ private void repaintIfBackgroundsDiffer(JComponent comp) {
             }
         }
     }
+
+    // Toggle flag for drawing the mnemonic state
+    private static boolean isMnemonicHidden = true;
+
+    /**
+     * Sets the state of the hide mnemonic flag. This flag is used by the
+     * component UI delegates to determine if the mnemonic should be rendered.
+     * This method is a non operation if the underlying operating system
+     * does not support the mnemonic hiding feature.
+     *
+     * @param hide true if mnemonics should be hidden
+     * @since 23
+     */
+    public static void setMnemonicHidden(boolean hide) {
+        if (UIManager.getBoolean("Button.showMnemonics")) {
+            // Do not hide mnemonics if the UI defaults do not support this
+            isMnemonicHidden = false;
+        } else {
+            isMnemonicHidden = hide;
+        }
+    }
+
+    /**
+     * Gets the state of the hide mnemonic flag. This only has meaning
+     * if this feature is supported by the underlying OS.
+     *
+     * @return true if mnemonics are hidden, otherwise, false
+     * @see #setMnemonicHidden
+     * @since 23
+     */
+    public static boolean isMnemonicHidden() {
+        if (UIManager.getBoolean("Button.showMnemonics")) {
+            // Do not hide mnemonics if the UI defaults do not support this
+            isMnemonicHidden = false;
```

Comment:
why does the get method also set the `isMnemonicHidden` flag? Shouldn't this only be done on the set method?

Reply:
> why does the get method also set the `isMnemonicHidden` flag? Shouldn't this only be done on the set method?

`isMnemonicHidden` is set to false to make sure if the UI defaults don't support to hide mnemonics, then don't hide it.

Reply:
if the UI defaults don't support hiding mnemonics, should a warning be given to the user?

Reply:
Don't think so.... as for Windows and Aqua the implementation is similar.